WMF Framework/Remote Notifications/Model/RemoteNotificationsModelController.swift

```diff
@@ -160,25 +150,32 @@ final class RemoteNotificationsModelController: NSObject {
     // inside the perform(_:) or the performAndWait(_:) methods.
     // https://developer.apple.com/documentation/coredata/using_core_data_in_the_background
     private func createNewNotification(from notification: RemoteNotificationsAPIController.NotificationsResult.Notification) {
-        guard let date = date(from: notification.timestamp?.utciso8601) else {
+        guard let date = date(from: notification.timestamp.utciso8601) else {
             assertionFailure("Notification should have a date")
             return
         }
-        guard let id = notification.id else {
-            assertionFailure("Notification must have an id")
-            return
-        }
 
-        let message = notification.message?.header?.wmf_stringByRemovingHTML()
+        let isRead = notification.readString == nil ? NSNumber(booleanLiteral: false) : NSNumber(booleanLiteral: true)
```

Comment:
Little question to clarify my understanding about `readString`/the JSON response's `read` property value: is it expected that the API returns a date string if a notification has been read (presumably the date the notification was marked as read), but nil for that property otherwise?

Reply:
@staykids this is pretty investigative, but yeah that's my understanding. The [notifications API](https://www.mediawiki.org/wiki/Notifications/API) doesn't have documentation on the response, but I've been pointed to [this](https://github.com/wikimedia/mediawiki-extensions-Echo/blob/66bb02fd8b8e74cd0277b20a26b000abcb35dcc2/includes/DataOutputFormatter.php#L134) area of code for determining what optional values to expect from here. If you follow that through ([here](https://github.com/wikimedia/mediawiki-extensions-Echo/blob/66bb02fd8b8e74cd0277b20a26b000abcb35dcc2/includes/model/Notification.php#L207) > [here](https://github.com/wikimedia/mediawiki-extensions-Echo/blob/4dd2a651e154fed4689ecdb7da6f3536a5c77f96/includes/model/Notification.php#L156) > [here](https://github.com/wikimedia/mediawiki-extensions-Echo/blob/37c602c5326ecd171b8eec3eedadade987cbdb26/includes/gateway/UserNotificationGateway.php#L75), it seems like the database table's `notification_read_timestamp` is set with the current date here within this `markRead` method, and that ultimately bubbles up to the API response value. I also found this comment [here](https://github.com/wikimedia/mediawiki-extensions-Echo/blob/839b833033a5cc4286ca0618b12188b670b87562/echo.sql#L43) that the `notification_read_timestamp` is the "Timestamp when the user read the notification, or null if unread".

Reply:
Wow great detective work 🕵️